packages/eslint-plugin/src/rules/no-suppress-xcss/index.ts

```diff
@@ -0,0 +1,79 @@
+import type { Rule } from 'eslint';
+
+function nodeIsTypeSuppressed(context: Rule.RuleContext, node: Rule.Node) {
+  if (!node.loc) {
+    return;
+  }
+
+  const comments = context.sourceCode.getAllComments();
+
+  for (const comment of comments) {
+    if (!comment.loc) {
+      continue;
+    }
+
+    const commentLine = comment.loc.start.line;
+    const nodeLine = node.loc.start.line;
+    const isCommentOnPreviousLine = nodeLine - 1 === commentLine;
+
+    if (
+      isCommentOnPreviousLine &&
+      ['@ts-expect-error', '@ts-ignore', '@ts-nocheck'].some((tag) => comment.value.includes(tag))
```

Comment:
Ah for context, `@ts-nocheck` applies to the entire file, so `isCommentOnPreviousLine` could be false and the type checking would still be suppressed

Reply:
I believe it's used at the top of the file? Would be good to double check

Reply:
Ah, fair shout. I'll check!